cvat-ui/src/components/invitation-watcher/invitation-watcher.tsx

```diff
@@ -0,0 +1,102 @@
+import Modal from 'antd/lib/modal';
+import { notification } from 'antd';
+
+const core = getCore();
+
+function InvitationWatcher(): JSX.Element {
+    const user = useSelector((state: CombinedState) => state.auth.user);
+    const history = useHistory();
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        const queryParams = new URLSearchParams(history.location.search);
+        const invitationKey = queryParams.get('invitation');
+
+        if (invitationKey) {
+            const storedInvitations = localStorage.getItem('invitations') || '[]';
+            const invitations = JSON.parse(storedInvitations);
+            if (!invitations.includes(invitationKey)) {
+                invitations.push(invitationKey);
+                localStorage.setItem('invitations', JSON.stringify(invitations));
+                localStorage.removeItem('currentOrganization');
+            }
+        }
+
+        if (user && user.isVerified) {
+            const storedInvitations = localStorage.getItem('invitations') || '[]';
+            const invitations = JSON.parse(storedInvitations);
+
+            const checkRequests = invitations.map(
+                (key: string) => core.organizations.invitation(key),
+            );
+            Promise.allSettled(checkRequests).then((promises) => {
+                const validInvitations = promises
+                    .map((promise) => (promise.status === 'fulfilled' ? promise.value : null))
+                    .filter((invitation) => !!invitation);
+
+                promises.forEach((promise) => {
+                    if (promise.status === 'rejected') {
+                        notification.error({
+                            message: 'Invitation error',
+                            className: 'cvat-invitation-error',
+                            description: promise.reason.toString(),
+                        });
+                    }
+                });
+
+                const expiredInvitations = validInvitations.filter((invitation) => invitation.expired);
+                expiredInvitations.forEach((invitation) => {
+                    notification.error({
+                        message: 'Invitation error',
+                        className: 'cvat-invitation-error',
+                        description: `The invitation to ${invitation.organization.slug} is expired. ` +
+                            'Please, contact organization owner.',
+                    });
+                });
+
+                const invitationsToBeShown = validInvitations.filter((invitation) => !invitation.expired);
+                if (invitationsToBeShown.length === 0) {
+                    localStorage.removeItem('invitations');
+                }
+                invitationsToBeShown.forEach((invitation: Invitation, index) => {
+                    const slug = invitation.organization instanceof Organization ? invitation.organization.slug : '';
+                    Modal.confirm({
```

Comment:
With forEach it will just show all the modals (I believe this is already not a good UX) at the same time, the latest with highest index probably, and after submitting the latest, it will remove all the other invitations from local storage and will update the page immediately